### PR TITLE
Optimized mkindex function in doc/conf.py and added two files in troubleshooting.

### PR DESCRIPTION
**Pull request by tuoping**
Optimized mkindex functions in doc/conf.py;
Changed from list of filenames to list of headlines in troubleshooting/index.md and development/index.md;
Added [The temperature undulates violently during early stages of MD](md-energy-undulation.md) in troubleshooting;
Added [Do we need to set rcut < half boxsize ?](rcut.md) in troubleshooting.

### CODE REVIEW

File: doc/troubleshooting/index.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
...
+[Installation](installation.md)
+
+[MD: cannot run LAMMPS after installing a new version of DeePMD-kit](md-version-compatibility.md)
+
+[Model compatability](model-compatability.md)
+
+[The temperature undulates violently during early stages of MD](md-energy-undulation.md)
+
+[Do we need to set rcut < half boxsize ?](rcut.md)
+
```
Comment (amcadmus) [inline doc/troubleshooting/index.md:15]: I would prefer the itemized presentation.

File: doc/troubleshooting/rcut.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Do we need to set rcut < half boxsize ?
+
+When seeking the neighbors of atom i under periodic boundary condition, deepmd-kit considers all j atoms within cutoff Rc from atom i in all mirror cells.
+
+So, no, so there is no limitation on the setting of rcut.
```
Comment (amcadmus) [inline doc/troubleshooting/rcut.md:5]: plz fix the typos in this sentence

File: doc/troubleshooting/index.md (hunk 1 above)
Comment (tuoping) [reply doc/troubleshooting/index.md:15]: OK

File: doc/troubleshooting/rcut.md (hunk 2 above)
Comment (tuoping) [reply doc/troubleshooting/rcut.md:5]: Ok, sorry.

**Pull request merged**